plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -0,0 +1,2073 @@
+            },
+            'ipv6': (match_delete_req_base +
+                     'match-prefix-set/config/openconfig-routing-policy-ext:ipv6-prefix-set'),
+            'local_preference': bgp_match_delete_req_base + 'config/local-pref-eq',
+            'metric': bgp_match_delete_req_base + 'config/med-eq',
+            'origin': bgp_match_delete_req_base + 'config/origin-eq',
+            'peer': (match_delete_req_base +
+                     'match-neighbor-set/config/'
+                     'openconfig-routing-policy-ext:address={}'.format(peer_str)),
+            'source_protocol': match_delete_req_base + 'config/install-protocol-eq',
+            'source_vrf': (match_delete_req_base +
+                           'openconfig-routing-policy-ext:match-src-network-instance'),
+            'tag': (match_delete_req_base +
+                    'match-tag-set/config/openconfig-routing-policy-ext:tag-value')
+        }
+
+        # Remove all appropriate "match" configuration for this route map if any of the
+        # following criteria are met:  (See the note below regarding what configuration
+        # is "appropriate"for deletion.)
+        #
+        # 1) Any top level attribute is specified with a value different from its current
+        #    configured value.
+        # 2) Any top level attribute is specified that is not currently configured.
+        # 3) The set of top level attributes specified does not include all currently
+        #    configured attributes (regardless of whether the specified values for
+        #    these attributes are the same as the ones courrently configured).
+        # (Note: Although the IPv6 attribute is defined as a nested dictionary
+        # to allow for future expansion, it is handled here as a top level
+        # attrbute because it currently has only one member.)
+        #
+        # When deletion has been triggered, an attribute is deleted only if it is
+        # not present at all in the requested configuration. (If it is present in
+        # the requested configuration, the "merge" phase of the "replaced" state
+        # operation will modify it as needed, so it doesn not need to be explicitly
+        # deleted during the "deletion" phase.)
+        #
+        cmd_top_level_key_set = set(cmd_match_keys).intersection(set(match_top_level_keys))
+        cfg_top_level_key_set = set(cfg_match_keys).intersection(set(match_top_level_keys))
+        symmetric_diff_set = cmd_top_level_key_set.symmetric_difference(cfg_top_level_key_set)
+        intersection_diff_set = cmd_top_level_key_set.intersection(cfg_top_level_key_set)
+        cmd_delete_dict = {}
+        if (cmd_top_level_key_set and symmetric_diff_set or
+                (any(keyname for keyname in intersection_diff_set if
+                     cmd_match_top[keyname] != cfg_match_top[keyname]))):
+            for cfg_match_key in cfg_match_top:
+                if cmd_match_top.get(cfg_match_key) is None:
```

Comment:
In the current implementation, the deletion is performed only for attributes that are present and not specified (i.e. have - want) and since the empty keys are removed, the attributes to be deleted could be obtained using `difference` as below:
```python
for cfg_match_key in cfg_top_level_set.difference(cmd_top_level_key_set):
    cmd_delete_dict[cfg_match_key] = cfg_match_top[cfg_match_key]
```

Reply:
Yes. Thanks for pointing that out. I initially implemented this with comparison of values for keys in the intersection_diff_set, and deletion only for a mismatch. After deciding, instead, not to delete any items that would be updated in the "modify" phase, I didn't think of simplifying the code as you have suggested here.

I have implemented this suggestion here and in the other two places that used a similar code sequence.

Note: The criteria for triggering the deletion are unchanged. Only the actual deletion logic, once triggered, can be simplified.

Reply:
Done.

(Just highlighting this to draw attention to re-checking and "Resolving" the issue.